pkg/kube/util/reference/reconciles.go

```diff
@@ -0,0 +1,95 @@
+package reference
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	crc "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"code.cloudfoundry.org/quarks-secret/pkg/kube/apis"
+	res "code.cloudfoundry.org/quarks-secret/pkg/kube/util/resources"
+	log "code.cloudfoundry.org/quarks-utils/pkg/ctxlog"
+)
+
+// ReconcileType lists all the types of reconciliations we can return,
+// for controllers that have types that can reference Secrets
+type ReconcileType int
+
+const (
+	// ReconcileForQuarksSecret represents the QuarksSecret CRD
+	ReconcileForQuarksSecret ReconcileType = iota
+)
+
+func (r ReconcileType) String() string {
+	return [...]string{
+		"QuarksSecret",
+	}[r]
+}
+
+// GetReconcilesWithFilter returns reconciliation requests for the QuarksSecret
+// that reference an object. The object can be a Secret, it accepts an admit function which is used for filtering the object
+func GetReconcilesWithFilter(ctx context.Context, client crc.Client, reconcileType ReconcileType, object apis.Object, versionCheck bool, admitFn func(v interface{}) bool) ([]reconcile.Request, error) {
+	objReferencedBy := func(parent interface{}) (bool, error) {
+		var (
+			objectReferences map[string]bool
+			err              error
+			name             string
+		)
+
+		switch object := object.(type) {
+		case *corev1.Secret:
```

Comment:
I'd get rid of the switch. This is never called for more than one type.

Reply:
Removed whole GetReconciles package.